diff --git a/CMakeLists.txt b/CMakeLists.txt
index f9505d2..c1f8cef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,8 @@ ament_package(
   CONFIG_EXTRAS "ament_cmake_export_include_directories-extras.cmake"
 )
 
+ament_package_config_install_dir(_config_install_dir)
 install(
-  DIRECTORY cmake
-  DESTINATION share/${PROJECT_NAME}
+  DIRECTORY cmake/
+  DESTINATION ${_config_install_dir}
 )
diff --git a/cmake/ament_export_include_directories.cmake b/cmake/ament_export_include_directories.cmake
index 9d21162..7bf0e36 100644
--- a/cmake/ament_export_include_directories.cmake
+++ b/cmake/ament_export_include_directories.cmake
@@ -37,7 +37,8 @@ macro(ament_export_include_directories)
       if(NOT IS_ABSOLUTE "${_arg}")
         # prefix relative paths with CMAKE_INSTALL_PREFIX
         # while avoiding to embed any absolute path
-        set(_arg "\${${PROJECT_NAME}_DIR}/../../../${_arg}")
+        ament_package_install_prefix(_prefix "\${${PROJECT_NAME}_DIR}/../../..")
+        set(_arg "${_prefix}/${_arg}")
         list_append_unique(_AMENT_EXPORT_RELATIVE_INCLUDE_DIRECTORIES "${_arg}")
       else()
         if(NOT IS_DIRECTORY "${_arg}")
