diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7671905..b4d78fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,7 +62,8 @@ endif()
 
 ament_package(CONFIG_EXTRAS "ament_cmake_gen_version_h-extras.cmake")
 
+ament_package_config_install_dir(_config_install_dir)
 install(
-  DIRECTORY cmake
-  DESTINATION share/${PROJECT_NAME}
+  DIRECTORY cmake/
+  DESTINATION ${_config_install_dir}
 )
